src/main/java/app/linguistai/bmvp/service/AccountService.java

```diff
@@ -192,39 +210,51 @@ public ResetToken generateEmailToken(String email) throws Exception {
 
             // create a new reset token
             ResetToken resetToken = new ResetToken(user);
+
+            log.info("Email token is generated for user with email {}.", email);
             
             return resetTokenRepository.save(resetToken);
-        } catch (Exception e) {
-            System.out.println("Email token generation exception for email");
+        } catch (NotFoundException e) {
+            log.error("User is not found for email {}", email);
             throw e;
+        } catch (Exception e) {
+            log.error("Generate email token failed for email {}", email, e);
+            throw new SomethingWentWrongException();
         }
     }
 
     public boolean validateResetCode(String email, String resetCode, boolean invalidate) throws Exception {
         try {
-            User user = accountRepository.findUserByEmail(email).orElse(null);
-            if (user == null) {
-                throw new NotFoundException("User with email [" + email + "] not found");
-            }
+            User user = accountRepository.findUserByEmail(email).orElseThrow(() -> new NotFoundException(User.class.getSimpleName(), true));
 
-            ResetToken resetToken = resetTokenRepository.findByUserAndResetCode(user, resetCode).orElse(null);
-            if (resetToken == null) {
-                throw new NotFoundException("Reset token for user with email [" + user.getEmail() + "] with code [" + resetCode + "] not found.");
-            }
+            ResetToken resetToken = resetTokenRepository.findByUserAndResetCode(user, resetCode).orElseThrow(() -> new NotFoundException("Reset token", true));
 
             if (!isResetTokenValid(resetToken)) {
-                return false;
+                throw new TokenException("Invalid password reset token");
```

Comment:
It looks like there are other functions depending on this method returning a boolean value, including `false` (i.e. `AccountController.saveResetPassword`, `AccountController.validateResetPassword`). Can you please check that no functionality is broken before merging? Thanks

Reply:
offffffffffffffffffffffffffffffffffffffffffffffffffff